coalib/results/Diff.py

```diff
@@ -166,16 +166,22 @@ def from_clang_fixit(cls, fixit, file):
         """
         assert isinstance(file, (list, tuple))
 
-        oldvalue = '\n'.join(file[fixit.range.start.line-1:
-                                  fixit.range.end.line])
-        endindex = fixit.range.end.column - len(file[fixit.range.end.line-1])-1
-
-        newvalue = (oldvalue[:fixit.range.start.column-1] +
-                    fixit.value +
-                    oldvalue[endindex:])
-        new_file = (file[:fixit.range.start.line-1] +
-                    type(file)(newvalue.splitlines(True)) +
-                    file[fixit.range.end.line:])
+        try:
+            oldvalue = '\n'.join(file[fixit.range.start.line-1:
+                                      fixit.range.end.line])
+            endindex = fixit.range.end.column - (
+                len(file[fixit.range.end.line-1])-1)
+
+            newvalue = (oldvalue[:fixit.range.start.column-1] +
+                        fixit.value +
+                        oldvalue[endindex:])
+            new_file = (file[:fixit.range.start.line-1] +
+                        type(file)(newvalue.splitlines(True)) +
+                        file[fixit.range.end.line:])
+        except LibclangError as err:
+            if 'libclang.so version' in str(err):
+                logging.error('coala requires clang 3.4')
+                raise err
```

Comment:
no that's not what I meant 😅 I meant doing something like this:
```
if libclang.version < 3.4:  # <-- this field probably is called differently
    raise LibclangError('libclang for clang 3.4')

# no try block
oldvalue = ...
...
```

Reply:
We could have used the __version__ attribute but it is not present in clang.cindex module.
I'll try to find a solution to this problem :) Thanks for your suggestion.

Reply:
Current approach is based on this: https://github.com/coala/coala/issues/3847#issuecomment-345529019

Reply:
Btw we may want to put the check into module scope, so we directly raise an exception, but not sure. Up to you to decide it. But yeah John is right, if we move the dependency into the bear that would be awesome, but not task of this PR ;)

Reply:
@Makman2  I tried finding a way so that we can do version check using `if libclang.version < 3.4` but was unable to find one, should we stick with the try block?
Or do you want me to change `libclang-py3~=3.4.0` to `libclang-py3=3.4` in requirements.txt ?